web/src/message_list_view.ts

```diff
@@ -383,7 +383,7 @@ function populate_group_from_message(
         const match_topic = util.get_match_topic(message);
         const stream_url = hash_util.by_stream_url(message.stream_id);
         const is_archived = stream_data.is_stream_archived(message.stream_id);
-        const topic_url = hash_util.by_stream_topic_url(message.stream_id, message.topic);
+        const topic_url = hash_util.by_channel_topic_permalink(message.stream_id, message.topic);
```

Comment:
Hmm, I think we will want this link to use the message ID that we already have here in the message recipient bar, not the one that `by_channel_topic_permalink` will look up.